packages/one-app-hmr/lib/server.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import express from 'express';
+import webpack from 'webpack';
+import webpackDevMiddleware from 'webpack-dev-middleware';
+import webpackHotMiddleware from 'webpack-hot-middleware';
+
+import { createHotModuleWebpackConfig } from './webpack';
+import { getStaticPath } from './webpack/utility';
+import { createStaticRenderMiddleware } from './html/middleware';
+import { createHotParrotMiddleware } from './parrot';
+import { createHotLanguagePacks } from './locale';
+
+export default function hmrServer({
+  port = 4000,
+  modules,
+  entryModule,
+  remoteModuleMap,
+  scenarios,
+  languagePacks,
+  useParrotMiddleware,
+  useLanguagePacks,
+} = {}) {
+  const app = express();
+
+  const webpackConfig = createHotModuleWebpackConfig({ modules, languagePacks });
+  const compiler = webpack(webpackConfig);
+  const devMiddleware = webpackDevMiddleware(compiler, {
+    index: false,
+    serverSideRender: true,
+    publicPath: webpackConfig.output.publicPath,
+    writeToDisk: true,
+    logTime: true,
+    reporter: () => undefined,
+  });
+  const hotMiddleware = webpackHotMiddleware(compiler, {
+    dynamicPublicPath: false,
+  });
+
+  app
+    .use('/static', express.static(getStaticPath()))
+    .use(devMiddleware)
+    .use(hotMiddleware);
+
+  if (useLanguagePacks && languagePacks.length > 0) {
+    createHotLanguagePacks(languagePacks, hotMiddleware.publish);
+  }
+
+  if (useParrotMiddleware && scenarios.length > 0) {
+    const parrotMiddleware = createHotParrotMiddleware(scenarios, hotMiddleware.publish);
+    app.use(parrotMiddleware);
+  }
+
+  app
+    .get(
+      '*',
+      createStaticRenderMiddleware({
+        entryModule,
+        remoteModuleMap,
+      })
+    )
+    .post('/error', (req, res) => {
+      res.status(202);
+    });
+
+  return [app, app.listen(port)];
```

Comment:
returns the `app` and `server`, in the event the library is used as a `node.js` API.